timemachine/fe/free_energy.py

```diff
@@ -1120,6 +1136,91 @@ def compute_dense():
     return U_kl
 
 
+def make_u_kl_fxn(trajs, initial_states):
+    """fxn(k, l) = "trajs[k] evaluated in ensembles[l]"
+
+    usage note: be careful of axis-ordering convention, see: https://github.com/proteneer/timemachine/issues/1100
+    """
+
+    summed_potentials = [make_summed_potential(s.potentials) for s in initial_states]
+
+    # validate assumption that initial states all have compatible potentials / ensembles
+    kBTs = [BOLTZ * state.integrator.temperature for state in initial_states]
```

Comment:
nit: could we just define a single `kBT = s_0.integrator.temperature` (below the assertions that immediately follow)?

Reply:
(same for `summed_potentials`)

Reply:
Weakly held style preference to leave this as-is unless there's a performance concern.

(also, if we later relax the definition of `assert_ensembles_compatible(a, b)` to allow  `a` and `b` to have different temperatures, this won't need to be updated)

Reply:
>was just initially confused by the implied generality

The implementation (`u_kn = u_k(x_n) / kBTs[k]`) is I think compatible with `len(set(kBTs)) > 1`, just the assertion `assert_ensembles_compatible` currently enforces `len(set(kBTs)) == 1`.

(For readability, we could add an assertion `assert len(set(kBTs)) == 1` or a comment to that effect, but I'd be inclined to leave the logic as-is.)

Reply:
(I guess instantiating the identical summed potentials does duplicate the system parameters, which could be a nontrivial chunk of memory for very large systems, but still probably not a concern)

Reply:
Ahh, that's a good point about the needless storage of the otherwise identical summed potentials, when we could just store `sp.params for sp in summed_potentials` -- didn't check how much additional space is occupied by a summed potential instance besides its parameter array, but have modified this to store an array of params rather than a list of summed_potentials in https://github.com/proteneer/timemachine/pull/1387/commits/4f5c0fc9e3a3f1591077dfcca6d59bdfd9bd07e4

Reply:
> didn't check how much additional space is occupied by a summed potential instance besides its parameter array

Quick check: `len(pickle.dumps(sp)) / len(pickle.dumps(sp.params))` is maybe around 2-3x on a couple vacuum and solvent examples